perceiver_pytorch/utils.py

```diff
@@ -0,0 +1,90 @@
+import torch
+import torch.nn.functional as F
+import numpy as np
+import math
+import einops
+
+
+def extract_image_patches(
+    x: torch.Tensor, kernel: int, stride: int = 1, dilation: int = 1
+) -> torch.Tensor:
+    """
+    Extract image patches in a way similar to TensorFlow extract_image_patches
+    Taken from https://discuss.pytorch.org/t/tf-extract-image-patches-in-pytorch/43837/8
+    :param x: Input Torch Tensor
+    :param kernel: Size of kernel
+    :param stride: Stride of patch
+    :param dilation: Dilation rate
+    :return: Tensor of size [Batch, Height, Width, Channels*kernel*stride]
+    """
+    # Do TF 'SAME' Padding
```

Comment:
Might be worth copy-and-pasting @jacobbieker's GitHub comment into a code comment here: "In the Perceiver JAX implementation they extract image patches matching TensorFlow's SAME padding. PyTorch doesn't have that same kind of option, so this is a way to do that."

Reply:
Done!